packages/enzyme-adapter-utils/src/Utils.js

```diff
@@ -262,23 +262,40 @@ export function getComponentStack(
 export function simulateError(
   error,
   catchingInstance,
+  catchingType,
   rootNode, // TODO: remove `rootNode` next semver-major
   hierarchy,
   getNodeType = nodeTypeFromType,
   getDisplayName = displayNameOfNode,
 ) {
-  const { componentDidCatch } = catchingInstance || {};
-  if (!componentDidCatch) {
+  const instance = catchingInstance || {};
+  const type = catchingType || {};
+
+  const { componentDidCatch } = instance;
+
+  const { getDerivedStateFromError } = type;
```

Comment:
i think this will work?
```suggestion
  const { getDerivedStateFromError } = {} = catchingType;
```

Reply:
I'll try this out.

Reply:
Ended up with `catchingType` at the end of params and thus with default value assigned already there.